src/main/scala/diplomacy/LazyModule.scala

```diff
@@ -27,10 +27,15 @@ abstract class LazyModule()(implicit val p: Parameters)
       m.getName != "children"
     }.flatMap { m =>
       if (classOf[LazyModule].isAssignableFrom(m.getReturnType)) {
-        Seq((m.getName, m.invoke(this)))
+        val obj = m.invoke(this)
+        if (obj eq null) Seq() else Seq((m.getName, obj))
```

Comment:
do you want to use Nil (http://www.scala-lang.org/api/2.7.0/scala/Nil$object.html) instead of Seq()?